hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaControllerManagement.java

```diff
@@ -615,6 +632,24 @@ private Action handleAddUpdateActionStatus(final JpaActionStatus actionStatus, f
         return savedAction;
     }
 
+    private String handleDownloadedActionStatus(final JpaActionStatus actionStatus, final JpaAction action) {
+        // information status entry - check for a potential DOS attack
+        assertActionStatusQuota(action);
+        assertActionStatusMessageQuota(actionStatus);
+
+        return isDownloadOnly(action) ? handleDownloadedActionStatusForDownloadOnlyAction(action) : null;
+    }
+
+    private String handleDownloadedActionStatusForDownloadOnlyAction(final JpaAction action) {
```

Comment:
Could this method be used for all action types (not just for download only) if we call action.setActive(false) and
target.setUpdateStatus(TargetUpdateStatus.IN_SYNC) only if the action is download only?